atlp-rwanda/script-heroes-bn-backend
# #174797508 Setting testing environment

#### What does this PR do?
Setting a testing environment
#### Description of Task to be completed?
- configuring mocha for running test
- configuring chai for assertions
- configuring chai-http for HTTP request during testing
#### How should this be manually tested?
- Clone the repo `git clone https://github.com/atlp-rwanda/script-heroes-bn-backend.git`
- Switch your working directory `cd script-heroes-bn-backend`
- Checkout the branch `git checkout ch-setup-testing-env-#174797508`
- Install all dependencies `npm install`
- Run the test `npm test`
> You should see two passing test one for landing endpoint and the other for not found handler
#### Any background context you want to provide?
* <ins>If it happens that mocha refuses to run the tests</ins>
     - Try installing it globally `npm i -g mocha` and run test again
#### What are the relevant pivotal tracker stories?
[#174797508](https://www.pivotaltracker.com/n/projects/2465784)
